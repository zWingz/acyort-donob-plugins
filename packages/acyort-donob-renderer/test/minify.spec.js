const fs = require('fs-extra')
const { join } = require('path')
const minify = require('../lib/minify')


describe('test minify', () => {
  const publicPath = join(__dirname, './fixtures/minify')
  const destPath = join(publicPath, '../dest')
  it('minify html && js && css', async () => {
    fs.copySync(publicPath, destPath)
    await minify(destPath)
    expect(fs.readFileSync(join(destPath, 'index.html'), {
      encoding: 'utf-8',
    })).toMatchSnapshot('minifiy:html')
    expect(fs.readFileSync(join(destPath, 'style.css'), {
      encoding: 'utf-8',
    })).toMatchSnapshot('minifiy:css')
    expect(fs.readFileSync(join(destPath, 'index.js'), {
      encoding: 'utf-8',
    })).toMatchSnapshot('minifiy:js')
    fs.removeSync(destPath)
  })
})
