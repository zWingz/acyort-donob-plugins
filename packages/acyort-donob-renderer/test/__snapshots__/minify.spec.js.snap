// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test minify minify html && js && css: minifiy:css 1`] = `"div{background:red;color:red}body{margin:0;padding:0}"`;

exports[`test minify minify html && js && css: minifiy:html 1`] = `"<html lang=\\"en\\"><head><meta charset=\\"UTF-8\\"><meta name=\\"viewport\\" content=\\"width=device-width,initial-scale=1\\"><meta http-equiv=\\"X-UA-Compatible\\" content=\\"ie=edge\\"><title>Document</title></head><body><div>text html minify</div></body></html>"`;

exports[`test minify minify html && js && css: minifiy:js 1`] = `"function test(){console.log(\\"test js uglisy\\")}test(),test(),test(),test();"`;
